File: statistic-service/src/main/java/com/statisticservice/model/EyeRecognitionModel.java
package com.statisticservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EyeRecognitionModel {
    private int id;
    private String modelLink;
    private String eyeModelName;
    private List<EyeRecognitionSampleHistory> eyeRecognitionSampleTrain;
    private float accuracy;
    private float f1_score;
    private float precision;
    private boolean isActive;
    private int epochs;
    private float learningRate;
    private int imageSize;
    private int batchSize;
    private String mappingLabel;
    private int trainingTime;
    private Date createDate;

}

File: ml-service/model/EyeRecognitionModel.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from .EyeRecognitionSampleHistory import EyeRecognitionSampleHistory
from .EyeRecognitionSample import EyeRecognitionSample
import uuid

class EyeRecognitionModel(BaseModel):
    id: Optional[str] = None
    modelLink: str = None
    eyeModelName: str  = None
    eyeRecognitionSampleHistory: list[EyeRecognitionSampleHistory]  = None
    accuracy: Optional[float] = None
    f1Score: Optional[float] = None
    precision: Optional[float] = None
    isActive: bool = True
    epochs: Optional[int] = None
    learningRate: Optional[float] = None
    imageSize: Optional[int] = None
    batchSize: Optional[int] = None
    mappingLabel: Optional[str] = None
    trainingTime:Optional[int] = None
    createDate: Optional[datetime] = None
    modelType: Optional[str] = None
   
    class Config:
        from_attributes = True
    def get_samples(self) -> list[EyeRecognitionSample]:
        """Lấy danh sách các sample từ history."""
        if not self.eyeRecognitionSampleHistory:
            return []
        return [history.eyeRecognitionSample for history in self.eyeRecognitionSampleHistory if history.eyeRecognitionSample]